### PR TITLE
API-39406-update-regex-change-address-city

### PR DESCRIPTION
**Pull request by stiehlrod**
## Summary

- Updates regex pattern for changeOfAddress.city & treatments.center.city. 
- Adds the minLength and maxLength to treatments.center.city. 
- Updates v2/dev/swagger.json & v2/production/swagger.json.

## Related issue(s)

- [API-39406](https://jira.devops.va.gov/browse/API-39406)

## Testing done

- [ ] *New code is covered by unit tests*
- Postman
request_body includes:
- place a # in the city name for treatment center & mailing address
- send over 30 characters or 0 for the treatment center city

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
	modified:   modules/claims_api/app/swagger/claims_api/v2/dev/swagger.json
	modified:   modules/claims_api/app/swagger/claims_api/v2/production/swagger.json
	modified:   modules/claims_api/config/schemas/v2/526.json
	modified:   modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb

## Acceptance criteria

- [X]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [ ]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [X]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/config/schemas/v2/526.json [hunk 1]
```diff
@@ -189,7 +189,7 @@
         "city": {
           "description": "City for the Veteran's new address.",
           "type": "string",
-          "pattern": "^$|^([-a-zA-Z'.' ])+$",
+          "pattern": "^$|([a-zA-Z0-9\\'-'.# ]([a-zA-Z0-9\\'-'.# ])?)+$",
           "maxLength": 1000,
```
Comment (tycol7) [inline modules/claims_api/config/schemas/v2/526.json:193]: Should we make `"minLength": 0` and `"maxLength": 30`?

File: modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb [hunk 2]
```diff
@@ -481,20 +481,16 @@
           end
         end
 
-        context 'when the city is invalid' do
+        context 'when the city is valid' do
```
Comment (tycol7) [inline modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb:484]: Should we also make a test in which the city is invalid? It seems like we are losing coverage.

File: modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb [hunk 3]
```diff
@@ -2239,13 +2235,13 @@ def update_json_and_submit(updated_json_lambda)
             context 'has invalid characters in it' do
               let(:treated_center_city) { 'LMNOP 6^7' }
```
Comment (tycol7) [inline modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb:2236]: Should we change this test name to something like `has valid special characters in it`? `invalid` seems misleading.

**Review by tycol7 — COMMENTED** [review]
Looking good! I think the ticket calls for min and max length requirements in `changeOfAddress.city`.

File: modules/claims_api/config/schemas/v2/526.json (hunk 1 above)
Comment (stiehlrod) [reply modules/claims_api/config/schemas/v2/526.json:193]: This ticket was changed while I was working on it and I did not know that the min and max length was added for this attribute as well. TY

File: modules/claims_api/config/schemas/v2/526.json [hunk 4]
```diff
@@ -189,8 +189,9 @@
         "city": {
           "description": "City for the Veteran's new address.",
           "type": "string",
-          "pattern": "^$|^([-a-zA-Z'.' ])+$",
-          "maxLength": 1000,
+          "pattern": "^$|([a-zA-Z0-9\\'-'.# ]([a-zA-Z0-9\\'-'.# ])?)+$",
+          "minLength": 0,
+          "maxLength": 30,
```
Comment (tycol7) [inline modules/claims_api/config/schemas/v2/526.json:194]: Do we need `"nullable": true` here? 🤷

File: modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb [hunk 5]
```diff
@@ -481,20 +481,28 @@
...
               json['data']['attributes']['changeOfAddress']['city'] = city
               data = json.to_json
               post submit_path, params: data, headers: auth_header
+              expect(response).to have_http_status(:accepted)
+            end
+          end
+        end
+
+        context 'when the city is invalid' do
+          it 'responds with 202' do
```
Comment (tycol7) [inline modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb:499]: nit: `422`

**Review by tycol7 — APPROVED** [review]
lgtm with one question

File: modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb (hunk 5 above)
Comment (stiehlrod) [reply modules/claims_api/spec/requests/v2/veterans/disability_compensation_request_spec.rb:499]: good catch

**Review by tycol7 — APPROVED** [review]

**Pull request merged**